pkg/controller/che/create.go

```diff
@@ -57,8 +58,9 @@ func (r *ReconcileChe) GenerateAndSaveFields(deployContext *deploy.DeployContext
 		}
 		if len(deployContext.CheCluster.Spec.Auth.IdentityProviderPostgresSecret) < 1 {
 			keycloakPostgresPassword := util.GeneratePasswd(12)
-			keycloakDeployment, err := r.GetEffectiveDeployment(deployContext.CheCluster, deploy.IdentityProviderName)
-			if err == nil {
+			keycloakDeployment := &appsv1.Deployment{}
+			exists, _ := deploy.GetNamespacedObject(deployContext, deploy.IdentityProviderName, keycloakDeployment)
```

Comment:
Are you sure that we should ignore error? maybe log it at least with level debug?